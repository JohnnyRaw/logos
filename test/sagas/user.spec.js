const mockSignInWithPopup = jest.fn();
const mockSignInWithCredential = jest.fn();
const mockConnectRoles = jest.fn();
const mockSignOut = jest.fn();

jest.mock('utils/firebaseClient', () =>
  ({
    connectRoles: mockConnectRoles,
    signInWithCredential: mockSignInWithCredential,
    signInWithPopup: mockSignInWithPopup,
    signOut: mockSignOut
  })
);

const { login, logout } = require('sagas/user');

describe('user', () => {
  it('login saga', () => {
    const generator = login({ payload: { provider: 'github' } });

    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next({
      credential: {},
      user: {}
    }).value).toMatchSnapshot();
    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().done).toBe(true);
  });

  it('logout saga', () => {
    const generator = logout();

    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().value).toMatchSnapshot();
    expect(generator.next().done).toBe(true);
  });
});
