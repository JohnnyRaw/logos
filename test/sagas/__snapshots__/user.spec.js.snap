exports[`user login saga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "github"
    ],
    "context": null,
    "fn": [Function mockConstructor]
  }
}
`;

exports[`user login saga 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function mockConstructor]
  }
}
`;

exports[`user login saga 3`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Object {
        "authenticated": true,
        "credential": Object {},
        "data": Object {
          "displayName": undefined,
          "email": undefined,
          "photoURL": undefined,
          "refreshToken": undefined,
          "uid": undefined
        }
      },
      "type": "USER_LOGIN_SUCCESS"
    },
    "channel": null
  }
}
`;

exports[`user login saga 4`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Object {
        "args": Array [
          Object {
            "pathname": "/cms",
            "search": undefined,
            "state": undefined
          }
        ],
        "method": "push"
      },
      "type": "@@router/CALL_HISTORY_METHOD"
    },
    "channel": null
  }
}
`;

exports[`user logout saga 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function mockConstructor]
  }
}
`;

exports[`user logout saga 2`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "type": "USER_LOGOUT_SUCCESS"
    },
    "channel": null
  }
}
`;

exports[`user logout saga 3`] = `
Object {
  "@@redux-saga/IO": true,
  "PUT": Object {
    "action": Object {
      "payload": Object {
        "args": Array [
          Object {
            "pathname": "/",
            "search": undefined,
            "state": undefined
          }
        ],
        "method": "push"
      },
      "type": "@@router/CALL_HISTORY_METHOD"
    },
    "channel": null
  }
}
`;
